Geometry_Engine/Create/ShapeProfile.cs

```diff
@@ -324,9 +324,101 @@ public static FabricatedISectionProfile FabricatedISectionProfile(double height,
 
         /***************************************************/
 
+        [Description("Creates a single FreeFormProfile from a collection of ICurves. \n" +
+                     "Checks if it's a valid profile and attempts to fix;  \n" + 
+                     " - coplanarity with eachother, by projecting onto the plane of the curve with the biggest area. \n" + 
+                     " - coplanarity with XYPlane, by rotating the curves to align. \n" + 
+                     " - curves on the XYPlane, by translating from one controlpoint to the origin. \n" +
+                     "Checks if it's a valid profile; If they're closed, Not zero area, curve curve intersections, selfintersections. ")]
         public static FreeFormProfile FreeFormProfile(IEnumerable<ICurve> edges)
         {
-            return new FreeFormProfile(edges);
+            IEnumerable<ICurve> result = edges.ToList();
+            
+            List<Point> cPoints = edges.SelectMany(x => x.IControlPoints()).ToList();
+
+            // Any Point not on the XY-Plane
+            if (cPoints.Any(x => Math.Abs(x.Z) > Tolerance.Distance))
+            {
+                // Is Planar
+                Plane plane = Compute.FitPlane(cPoints);
+                if (cPoints.Any(x => x.Distance(plane) > Tolerance.Distance))
```

Comment:
The content in here is not supported yet for nurbs. If this happens, and we should exit out this larger if statement, as the orientation methods below wont give predictable results if this happens.